utils/zerovec/src/yoke_impls.rs

```diff
@@ -221,16 +221,16 @@ where
         unsafe {
             // Similar problem as transform(), but we need to use ptr::read since
             // the compiler isn't sure of the sizes
-            let ptr: *const Self::Output = (&self as *const Self).cast();
-            mem::forget(self);
+            let this = mem::ManuallyDrop::new(self);
+            let ptr: *const Self::Output = (&*this as *const Self).cast();
             ptr::read(ptr)
         }
     }
     #[inline]
     unsafe fn make(from: Self::Output) -> Self {
         debug_assert!(mem::size_of::<Self::Output>() == mem::size_of::<Self>());
```

Comment:
unrelated, but you should be able to assert this at compile time:
```suggestion
        const _: () = assert!(mem::size_of::<Self::Output>() == mem::size_of::<Self>());
```

Reply:
Ah right, I though this was changed in 1.61 but it doesn't work.